### PR TITLE
Document undocumented properties

### PR DESCRIPTION
**Pull request by bergie**
Handles various low-hanging issues by adding missing properties where needed.

### CODE REVIEW

**Comment by bergie** [comment]
@jonnor please review

File: schemata/item.yaml [hunk 1]
```diff
@@ -31,4 +38,6 @@ properties:
     - { type: 'null' }
     - { type: 'string', format: 'date-time' }
 
-required: [content]
+anyOf:
```
Comment (jonnor) [inline schemata/item.yaml]: Shouldn't this be `oneOf`? Must either have `content` or `starred`, right?

File: schemata/share.yaml [hunk 2]
```diff
@@ -52,3 +52,6 @@ properties:
         example: Bergie Today
     required: [url]
 required: [url]
+dependencies:
```
Comment (jonnor) [inline schemata/share.yaml:55]: Nice, did not know that feature before.

Comment (jonnor) [reply schemata/share.yaml:55]: https://spacetelescope.github.io/understanding-json-schema/reference/object.html#dependencies

File: schemata/item.yaml (hunk 1 above)
Comment (bergie) [reply schemata/item.yaml]: You're correct. Let me push a fix

File: schemata/siteconfig.yaml [hunk 3]
```diff
@@ -146,7 +148,7 @@ properties:
     minimum: 0.0
     maximum: 1.0
   site_lang:
-    type: string
+    type: [string, "null"]
```
Comment (jonnor) [inline schemata/siteconfig.yaml:151]: Bit annoying to allow `null` everywhere?

File: schemata/share.yaml (hunk 2 above)
Comment (bergie) [reply schemata/share.yaml:55]: Me neither. But trying to get the validation to be as close as possible to reality made me peruse the JSON schema docs and find bunch of goodies.

Ideally we should get these to stage where schemas can be trusted for all other validation except side effects (UUID references matching database etc). But that will probably take a few iterations

File: schemata/siteconfig.yaml (hunk 3 above)
Comment (bergie) [reply schemata/siteconfig.yaml:151]: @jonnor yeah, but otherwise validation fails for `nullable` properties.

**Comment by jonnor** [comment]
Looking good.

**Comment by jonnor** [comment]
@bergie Merged, deploy at will

**Pull request merged**